frontend/public/components/secrets/create-secret.tsx

```diff
@@ -167,11 +199,93 @@ class WebHookSecretSubform extends React.Component<WebHookSecretSubformProps, We
   }
 }
 
+class SourceSecretSubform extends React.Component<SourceSecretSubformProps, SourceSecretSubformState> {
+  constructor (props) {
+    super(props);
+    this.state = this.state = {
+      authenticationType: SecretType.basicAuth,
+      stringData: {
+        username: '',
+        password: '',
+        privateKey: '',
+      },
+    };
+    this.changeData = this.changeData.bind(this);
+    this.changeAuthenticationType = this.changeAuthenticationType.bind(this);
+  }
+  changeAuthenticationType(event) {
+    this.setState({
+      authenticationType: event.target.value
+    }, () => this.props.onChange(this.state));
+  }
+  changeData(event) {
+    const updatedData = _.assign(this.state.stringData, {[event.target.name]: event.target.value});
+    this.setState({
+      stringData: updatedData
+    }, () => this.props.onChange(this.state));
+  }
+  render () {
+    const basicAuthSubform = <React.Fragment>
+      <div className="form-group">
+        <label className="control-label" htmlFor="username">Username</label>
+        <div className="modal-body__field">
+          <input className="form-control" id="username" type="text" name="username" onChange={this.changeData} value={this.state.stringData.username} required/>
+          <p className="help-block text-muted">Optional username for Git authentication.</p>
+        </div>
+      </div>
+      <div className="form-group">
+        <label className="control-label" htmlFor="password">Password or Token</label>
+        <div className="modal-body__field">
+          <input className="form-control" id="password" type="password" name="password" onChange={this.changeData} value={this.state.stringData.password} required/>
+          <p className="help-block text-muted">Password or token for Git authentication. Required if a ca.crt or .gitconfig file is not specified.</p>
+        </div>
+      </div>
+    </React.Fragment>;
+
+    const sshAuthSubform = <div className="form-group">
+      <label className="control-label" htmlFor="private-key">SSH Private Key</label>
+      <div className="modal-body__field">
+        <div className="input-group">
+          <input type="text" className="form-control" readOnly disabled/>
+          <span className="input-group-btn">
+            <span className="btn btn-default btn-file">
+              Browse&hellip;
```

Comment:
Is `Browse&hellip;` correct?

Reply:
It should be -> https://github.com/openshift/origin-web-console/blob/efea73189b2e79f8923279b0935374c47d760942/app/views/directives/osc-file-input.html#L16